test/lint/lint-git-commit-check.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Copyright (c) 2017-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Linter to check that commit messages have a new line before the body
+# or no body at all
+#
+# reference: gist.github.com/agnivade/67b42d664ece2d4210c7
+#
+# WARNING: Using a "+" character in commit messages can result in 
+# improper parsing
+
+export LC_ALL=C
+while read -d "+" -r msg  || [[ -n "$msg" ]]; do
```

Comment:
This "+" idea is clever!  However, I don't think it's one we can live with.  The "+" character is actually used surprisingly often in the bodies of commit messages.

`git log --grep='+' --oneline | wc -l`
\> 706

Two suggested solutions, if you're looking for somewhere to start
- Try and find a list of valid characters in commit messages, and see if you can use one that isn't in the list.  Maybe "#" would work?  I know that emojis work in commit messages, so maybe this isn't a viable option, but check it out :)
 - Try iterate over each commit, as opposed to getting them all at once.

Reply:
After looking at the script some more, I decided to iterate over each commit by hash value in order to circumvent this delimiter problem. Thank you for the suggestion.